pkg/ebpf/c/common/context.h

```diff
@@ -21,75 +21,53 @@ statfunc void reset_event_args(program_data_t *);
     struct task_struct *leader = get_leader_task(task);
     struct task_struct *up_parent = get_leader_task(get_parent_task(leader));
 
     // Task Info on Host
-    context->task.host_tid = id;
-    context->task.host_pid = id >> 32;
-    context->task.host_ppid = get_task_pid(up_parent); // always a real process (not a lwp)
+    tsk_ctx->host_ppid = get_task_pid(up_parent); // always a real process (not a lwp)
     // Namespaces Info
-    context->task.tid = get_task_ns_pid(task);
-    context->task.pid = get_task_ns_tgid(task);
+    tsk_ctx->tid = get_task_ns_pid(task);
+    tsk_ctx->pid = get_task_ns_tgid(task);
 
     u32 task_pidns_id = get_task_pid_ns_id(task);
     u32 up_parent_pidns_id = get_task_pid_ns_id(up_parent);
 
     if (task_pidns_id == up_parent_pidns_id)
-        context->task.ppid = get_task_ns_pid(up_parent); // e.g: pid 1 will have nsppid 0
+        tsk_ctx->ppid = get_task_ns_pid(up_parent); // e.g: pid 1 will have nsppid 0
 
-    context->task.pid_id = task_pidns_id;
-    context->task.mnt_id = get_task_mnt_ns_id(task);
+    tsk_ctx->pid_id = task_pidns_id;
+    tsk_ctx->mnt_id = get_task_mnt_ns_id(task);
     // User Info
-    context->task.uid = bpf_get_current_uid_gid();
+    tsk_ctx->uid = bpf_get_current_uid_gid();
     // Times
-    context->task.start_time = get_task_start_time(task);
-    context->task.leader_start_time = get_task_start_time(leader);
-    context->task.parent_start_time = get_task_start_time(up_parent);
+    tsk_ctx->start_time = get_task_start_time(task);
+    tsk_ctx->leader_start_time = get_task_start_time(leader);
+    tsk_ctx->parent_start_time = get_task_start_time(up_parent);
 
-    context->task.flags = 0;
+    tsk_ctx->flags = 0;
 
     if (is_compat(task))
-        context->task.flags |= IS_COMPAT_FLAG;
+        tsk_ctx->flags |= IS_COMPAT_FLAG;
 
     // Program name
-    __builtin_memset(context->task.comm, 0, sizeof(context->task.comm));
-    ret = bpf_get_current_comm(&context->task.comm, sizeof(context->task.comm));
-    if (unlikely(ret < 0)) {
-        tracee_log(ctx, BPF_LOG_LVL_ERROR, BPF_LOG_ID_GET_CURRENT_COMM, ret);
-        return -1;
-    }
+    bpf_get_current_comm(&tsk_ctx->comm, sizeof(tsk_ctx->comm));
 
     // UTS Name
     char *uts_name = get_task_uts_name(task);
-    if (uts_name) {
-        __builtin_memset(context->task.uts_name, 0, sizeof(context->task.uts_name));
-        bpf_probe_read_str(&context->task.uts_name, TASK_COMM_LEN, uts_name);
-    }
+    if (uts_name)
+        bpf_probe_read_str(&tsk_ctx->uts_name, TASK_COMM_LEN, uts_name);
 
     // Cgroup ID
     if (options & OPT_CGROUP_V1) {
-        context->task.cgroup_id = get_cgroup_v1_subsys0_id(task);
+        tsk_ctx->cgroup_id = get_cgroup_v1_subsys0_id(task);
     } else {
-        context->task.cgroup_id = bpf_get_current_cgroup_id();
+        tsk_ctx->cgroup_id = bpf_get_current_cgroup_id();
     }
 
-    // Context timestamp
```

Comment:
All the four below will be enriched in userspace? Or is there another reason for removing them?

Reply:
exactly - they all moved to `init_program_data() ` but stack_id, which is set in the submit stage anyway